packages/ec-core/src/operator/selector/mod.rs

```diff
@@ -30,8 +81,159 @@ where
     ) -> Result<&'pop P::Individual, Self::Error>;
 }
 
+/// A wrapper that converts a `Selector` into an `Operator`.
+///
+/// This allows the inclusion of `Selector`s in chains of operators.
+///
+/// See [the `operator` module docs](crate::operator#wrappers) for more on the
+/// design decisions behind using wrappers.
+///
+/// # Examples
+///
+/// Here we illustrate the implementation of a simple [`Selector`],
+/// `First`, which returns the first item in a vector. We then wrap that in a
+/// [`Select`] to create an operator. Then calling [`Operator::apply`] on the
+/// operator is the same as calling [`Selector::select`] directly on the
+/// selector.
+///
+/// ```
+/// # use anyhow::Context;
+/// # use rand::rngs::ThreadRng;
+/// #
+/// # use ec_core::operator::selector::{Select, Selector};
+/// # use ec_core::operator::Operator;
+/// #
+/// struct First; // Simple selector that always returns the first element in a vector.
+///
+/// impl<T> Selector<Vec<T>> for First {
+///     type Error = anyhow::Error;
+///
+///     fn select<'pop>(
+///         &self,
+///         population: &'pop Vec<T>,
+///         _: &mut ThreadRng,
+///     ) -> anyhow::Result<&'pop T> {
+///         population
+///             .first()
+///             .context("Should be at least one individual in the population")
+///     }
+/// }
+///
+/// // Create a `Select` operator from the `First` selector
+/// let select = Select::new(First);
+/// let population: Vec<u8> = vec![5, 8, 9];
+/// // Selectors return references to the individuals they choose, so we
+/// // get a `&u8` back from `select` and `apply`.
+/// let selector_result: &u8 = First.select(&population, &mut rand::thread_rng()).unwrap();
+/// assert_eq!(*selector_result, 5);
+/// let operator_result: &u8 = select.apply(&population, &mut rand::thread_rng()).unwrap();
+/// assert_eq!(selector_result, operator_result);
```

Comment:
maybe also add some empty lines in between this huge block for visual separation? Just kinda noticed this seemed pretty bunched up in comparision with our other code.